packages/graphql/src/services/courses.ts

```diff
@@ -93,16 +99,41 @@ export async function joinCourse(
 
   if (!participation) return null
 
+  const lbEntry = await ctx.prisma.leaderboardEntry.create({
```

Comment:
create the leaderboard entry for the course as soon as participant joins the course. this prevents issues down the line and ensures we can simply update later on, instead of update or create on getting a question response anywhere.